File: week06/explore_phila_data/load_phl_pwd_parcels.mjs
import dotenv from 'dotenv';
dotenv.config();

import { BigQuery } from '@google-cloud/bigquery';

const bucketName = process.env.DATA_LAKE_BUCKET;
const datasetName = process.env.DATA_LAKE_DATASET;

// Load the data into BigQuery as an external table
const preparedBlobname = 'prepared/phl_pwd_parcels/phl_pwd_parcels.jsonl';
const tableName = 'phl_pwd_parcels';
const tableUri = `gs://${bucketName}/${preparedBlobname}`;

const createTableQuery = `
CREATE OR REPLACE EXTERNAL TABLE \`${datasetName}.${tableName}\`
OPTIONS (
  format = 'JSON',
  uris = ['${tableUri}']
)
`;

const bigqueryClient = new BigQuery();
await bigqueryClient.query(createTableQuery);
console.log(`Loaded ${tableUri} into ${datasetName}.${tableName}`);

File: week06/explore_phila_data/load_phl_pwd_parcels.py
from dotenv import load_dotenv
load_dotenv()

import os
from google.cloud import bigquery

bucket_name = os.getenv('DATA_LAKE_BUCKET')
dataset_name = os.getenv('DATA_LAKE_DATASET')

# Load the data into BigQuery as an external table
prepared_blobname = 'prepared/phl_pwd_parcels/prepared_phl_pwd_parcels.jsonl'
table_name = 'phl_pwd_parcels'
table_uri = f'gs://{bucket_name}/{prepared_blobname}'

create_table_query = f'''
CREATE OR REPLACE EXTERNAL TABLE {dataset_name}.{table_name}
OPTIONS(
  format='json',
  uris=['{table_uri}']
)
'''

bigquery_client = bigquery.Client()
bigquery_client.query_and_wait(create_table_query)
print(f'Loaded {table_uri} into {dataset_name}.{table_name}')
